src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -380,6 +380,8 @@ def _(self, node: QuantumGate) -> None:
         qubit_lengths = np.array(
             [self.context.qubit_mapping.get_qubit_size(qubit) for qubit in qubits]
         )
+        if any(not isinstance(qubit_length, np.integer) for qubit_length in qubit_lengths):
+            raise NotImplementedError("Undetermined qubit register size")
```

Comment:
We run into issues with `cnot qs[0:n1], qs[0:n2];` as we can't verify whether these registers are the same size and furthermore the logic immediately below this won't know how many gates to create and what symbols to use for each.

We can remove this constraint once we support a symbolic "loop" construct in the simulator IR